py-polars/tests/unit/io/test_cloud.py

```diff
@@ -11,3 +52,53 @@ def test_err_on_s3_glob() -> None:
         pl.scan_parquet(
             "s3://saturn-public-data/nyc-taxi/data/yellow_tripdata_2019-1.*.parquet"
         )
+
+
+@pytest.mark.usefixtures("_s3")
+def test_read_csv_on_s3() -> None:
+    df = pl.read_csv(
+        "s3://bucket/foods1.csv",
+        storage_options={"endpoint_url": f"http://{host}:{port}"},
+    )
+    assert df.columns == ["category", "calories", "fats_g", "sugars_g"]
+    assert df.shape == (27, 4)
+
+
+@pytest.mark.usefixtures("_s3")
+def test_read_ipc_on_s3() -> None:
+    df = pl.read_ipc(
+        "s3://bucket/foods1.ipc",
+        storage_options={"endpoint_url": f"http://{host}:{port}"},
+    )
+    assert df.columns == ["category", "calories", "fats_g", "sugars_g"]
+    assert df.shape == (27, 4)
+
+
+@pytest.mark.usefixtures("_s3")
+def test_scan_ipc_on_s3() -> None:
+    df = pl.scan_ipc(
+        "s3://bucket/foods1.ipc",
+        storage_options={"endpoint_url": f"http://{host}:{port}"},
+    )
+    assert df.columns == ["category", "calories", "fats_g", "sugars_g"]
+    assert df.collect().shape == (27, 4)
+
+
+@pytest.mark.usefixtures("_s3")
+def test_read_parquet_on_s3() -> None:
+    df = pl.read_parquet(
+        "s3://bucket/foods1.parquet",
+        storage_options={"endpoint_url": f"http://{host}:{port}"},
+    )
+    assert df.columns == ["category", "calories", "fats_g", "sugars_g"]
+    assert df.shape == (27, 4)
+
+
+@pytest.mark.usefixtures("_s3")
+def test_scan_parquet_on_s3() -> None:
+    df = pl.scan_parquet(
+        "s3://bucket/foods1.parquet",
+        storage_options={"endpoint_url": f"http://{host}:{port}"},
+    )
+    assert df.columns == ["category", "calories", "fats_g", "sugars_g"]
+    assert df.collect().shape == (27, 4)
```

Comment:
Seems `scan_*` are yet to be working properly; tests succeeded on schema reading part, but failed on `.collect()` part. This is not `moto`'s fault, `scan_parquet` currently fails on master too.

(try `pl.scan_parquet("s3://saturn-public-data/nyc-taxi/data/yellow_tripdata_2019-01.parquet").collect()` and you will see)